src/sentry/issues/highlights.py

```diff
@@ -0,0 +1,72 @@
+from collections.abc import Mapping
+from typing import TypedDict
+
+from jsonschema import Draft7Validator
+from jsonschema.exceptions import ValidationError as SchemaValidationError
+from rest_framework import serializers
+from rest_framework.serializers import ValidationError
+
+from sentry.models.project import Project
+from sentry.utils.json import JSONData
+from sentry.utils.platform_categories import BACKEND, FRONTEND, MOBILE
+
+HIGHLIGHT_CONTEXT_SCHEMA: JSONData = {
```

Comment:
I would prefer if we did not use JSON schema validation here, as it is not consistent with how we do validation with our other serializers. maybe something like the below could work?


```
from rest_framework import serializers
import re

class PatternPropertiesField(serializers.Field):
    def to_internal_value(self, data):
        if not isinstance(data, dict):
            raise serializers.ValidationError("Expected a dictionary.")
        
        for key, value in data.items():
            if not re.match(r'^.*$', key):
                raise serializers.ValidationError(f"Key {key} is invalid.")
            if not isinstance(value, list) or not all(isinstance(item, str) for item in value):
                raise serializers.ValidationError(f"Value for {key} must be a list of strings.")
        
        return data

    def to_representation(self, value):
        return value

class HighlightContextSerializer(serializers.Serializer):
    data = PatternPropertiesField()

    def validate(self, attrs):
        data = attrs.get('data', {})
        if any(key for key in data if not re.match(r'^.*$', key)):
            raise serializers.ValidationError("All keys must match the specified pattern.")
        return attrs
```

Reply:
Implemented the above, thanks for that, I agree that it's more consistent this way. I only implemented the field and used in the Detailed Project Serializer.